src/BitStringAddresses/bosefs.jl

```diff
@@ -327,36 +325,60 @@ end
         result += bosons
         chunk >>>= bosons % UInt
     end
-    return (result, orbital, bit_position), (i, chunk, bits_left, orbital)
+    return BoseFSIndex(result, orbital, bit_position), (i, chunk, bits_left, orbital)
 end
 
-"""
-    BoseFS2C{NA,NB,M,AA,AB} <: AbstractFockAddress
+function find_site(b::BoseFS, index)
```

Comment:
Can we please rename `find_site`? Sites really only make sense in lattices and mode or orbital are more general concepts. I'd like to eliminate lattice jargon in the code. I realise, however, that we have mode, as in `num_modes` and orbital as in `occupied_orbitals` as synonymous terms. Maybe it would be better to consistently just use of them, maybe `mode`?
```suggestion
function find_mode(b::BoseFS, index)
```